test/run-webhook.py

```diff
@@ -0,0 +1,285 @@
+    {webhook_config} \
+    {cri_args} \
+    {helm_args} \
+    --debug\
+    ".format(chart_name=HELM_CHART_NAME,
+             location=helm_chart_location,
+             namespace=NAMESPACE,
+             webhook_config=get_webhook_helm_config(),
+             cri_args=cri_args,
+             helm_args=extra_helm_args)
+    print("Helm command: {}".format(helm_install_command))
+    os.system(helm_install_command)
+
+    res = False
+    try:
+        res = do_test()
+    except Exception as e:
+        print(e)
+        res = False
+    finally:
+        # Best effort cleanup work
+        try:
+            # Save Agent and controller logs
+            shared_test_code.save_agent_and_controller_logs(
+                namespace=NAMESPACE)
+        finally:
+            # Delete akri and check that controller and Agent pods deleted
+            os.system("\
+                helm delete {chart_name} \
+                --namespace={namespace}\
+                ".format(
+                chart_name=HELM_CHART_NAME,
+                namespace=NAMESPACE,
+            ))
+            # Delete Akri CRDs
+            client.ApiextensionsV1Api().delete_custom_resource_definition(
+                "{kind}.{group}".format(kind="configurations", group=GROUP),
+                body=client.V1DeleteOptions())
+            client.ApiextensionsV1Api().delete_custom_resource_definition(
+                "{kind}.{group}".format(kind="instances", group=GROUP),
+                body=client.V1DeleteOptions())
+            # Delete Webhook Secret
+            client.CoreV1Api().delete_namespaced_secret(name=WEBHOOK_NAME,
+                                                        namespace=NAMESPACE)
+            if res:
+                # Only test cleanup if the test has succeeded up to now
+                if not shared_test_code.check_akri_state(
+                        0, 0, 0, 0, 0, 0, namespace=NAMESPACE):
+                    print(
+                        "Akri not running in expected state after helm delete")
+                    raise RuntimeError("Scenario Failed")
+
+    if not res:
+        raise RuntimeError("Scenario Failed")
+
+
+def do_test() -> bool:
+    kubeconfig_path = shared_test_code.get_kubeconfig_path()
+    print("Loading k8s config: {}".format(kubeconfig_path))
+    config.load_kube_config(config_file=kubeconfig_path)
+
+    # Get kubectl command
+    kubectl_cmd = shared_test_code.get_kubectl_command()
+
+    # Ensure Helm Akri installation applied CRDs and set up Agent and Controller
+    print("Checking for CRDs")
+    if not shared_test_code.crds_applied():
+        print("CRDs not applied by helm chart")
+        return False
+
+    print("Checking for initial Akri state")
+
+    if not shared_test_code.check_akri_state(1, 1, 0, 0, 0, 0):
+        print("Akri not running in expected state")
+        os.system("sudo {} get pods,services,akric,akrii --show-labels".format(
+            kubectl_cmd))
+        return False
+
```

Comment:
might be interesting to validate/query the webhook pod/svc here to see if it is running?

Reply:
something like `os.system("sudo {} get pods,services --show-labels".format(kubectl_cmd))` ... or something with more helpful info (`-o yaml`?)

Reply:
Yes, this is actually quite an interesting challenge, I've not worked against such opaque cluster(s) before.

Debugging added: https://github.com/deislabs/akri/pull/206/commits/5e82996e4a005569788410cc80ce945174eb22fd

Reply:
Apologies for polluting your build history with my yellows and reds; it would be useful to be able to deploy to "pet" VMs (perhaps even on a different cloud) so that I could ssh into the VM to see what's going on (and not run up your bill).

Reply:
no worries ... you should see some of the days and days and days and days of failures i've created in our actions history.  lol.

Reply:
i'm not seeing the output of the get commands ... you might need sudo?

Reply:
Ah! That was perplexing me, thanks.

It's that time of the day where my dog and I go for another walk, will resume tomorrow. Eesh 😁

Reply:
looks like sudo at least helped generate some output.  you can see the deployment, service, etc.  I see `Deployment does not have minimum availability` in the webhook deployment status and am not sure what it exactly it means, but it doesn't sound warm and fuzzy.

Reply:
Thanks @bfjelds!

I'm (finally) back online (curtailed working day in pretirement :smile: ) and on it!

I don't understand why `sudo` was necessary to get stdout. Perhaps I should have used the Kubernetes SDK instead of `os.system` (perhaps `subprocess` instead of `os`?)? Anyway...

I plan to stub the matrix this morning to reduce the Actions burden each commit.